projects/driver_controls/src/pedal_fsm.c

```diff
@@ -0,0 +1,81 @@
+#include "pedal_fsm.h"
+#include "input_event.h"
+#include "event_arbiter.h"
+#include "log.h"
+
+// Pedal FSM state definitions
+
+FSM_DECLARE_STATE(state_brake);
+FSM_DECLARE_STATE(state_coast);
+FSM_DECLARE_STATE(state_driving);
+FSM_DECLARE_STATE(state_cruise_control);
+
+// Pedal FSM transition table definitions
+
+FSM_STATE_TRANSITION(state_brake) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_COAST, state_coast);
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_PRESSED, state_driving);
+}
+
+FSM_STATE_TRANSITION(state_coast) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_COAST, state_coast);
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_PRESSED, state_driving);
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_BRAKE, state_brake);
+  FSM_ADD_TRANSITION(INPUT_EVENT_CRUISE_CONTROL, state_cruise_control);
+}
+
+FSM_STATE_TRANSITION(state_driving) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_BRAKE, state_brake);
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_COAST, state_coast);
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_PRESSED, state_driving);
+  FSM_ADD_TRANSITION(INPUT_EVENT_CRUISE_CONTROL, state_cruise_control);
+}
+
+FSM_STATE_TRANSITION(state_cruise_control) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_MECHANICAL_BRAKE, state_brake);
+  FSM_ADD_TRANSITION(INPUT_EVENT_PEDAL_BRAKE, state_brake);
+  FSM_ADD_TRANSITION(INPUT_EVENT_CRUISE_CONTROL, state_brake);
+  FSM_ADD_TRANSITION(INPUT_EVENT_CRUISE_CONTROL_INC, state_cruise_control);
+  FSM_ADD_TRANSITION(INPUT_EVENT_CRUISE_CONTROL_DEC, state_cruise_control);
+}
+
+// Pedal FSM arbiter function
+
+static bool prv_check_pedal(const Event *e) {
+  return true;
+}
+
+// Pedal FSM output functions
+
+static void prv_state_output(FSM *fsm, const Event *e, void *context) {
+  EventArbiterCheck *event_check = fsm->context;
+  *event_check = prv_check_pedal;
+
+  StateTransition current_state;
```

Comment:
`State` was defined as a struct, so two states can't be compared directly. And since the `StateTransition` function pointer is unique among states, the comparisons could be done with this member.